### ISSUE TITLE
Multi-label output with Softmax

### ISSUE THREAD

**Issue by Petterhg** [apache/incubator-mxnet#5514]
Is it possible to make a Softmax prob. output for a dataset of the following structure:

data_shape(1, 3) : Car_model, Color, Manufacturer
label_shape(1, 5) : Part1, Part2, Part3, Part4, Part5

**So, I want to have the probability of a part given a car model. That is, a vector of probabilities of each part given some features of a car. That is a the output of the network should be a matrix with size (#datapoints X width of label_shape). This is the code so far (I'm getting dimension issues)**:

`const batch_size = 5
data_provider = mx.CSVIter(data_csv=CSV_DATA_PATH, data_shape=(1,3), label_csv=CSV_LABEL_PATH, label_shape=(1,5), batch_size=batch_size)
eval_provider = mx.CSVIter(data_csv=CSV_EVAL_DATA_PATH, data_shape=(1,3), label_csv=CSV_EVAL_LABEL_PATH, label_shape=(1,5), batch_size=batch_size)
data = mx.Variable(:data)
label = mx.Variable(:softmax_label)

#Network design...
fc1  = mx.FullyConnected(data, name=:fc1, num_hidden=50)
bn1  = mx.BatchNorm(fc1)
act1 = mx.Activation(bn1, name=:relu1, act_type=:relu)
fc2  = mx.FullyConnected(act1, name=:fc2, num_hidden=5)
softmax = mx.SoftmaxOutput(fc2, label, name=:softmax)

model = mx.FeedForward(softmax, context=mx.cpu())
optimizer = mx.SGD(lr=learning_rate, momentum=momentum, weight_decay=weight_decay)

mx.fit(model, optimizer, data_provider, n_epoch=N_EPOCH, 
    eval_data=eval_provider,  initializer=mx.UniformInitializer())`

Output is something like:

**MXNet.mx.MXError("Error in operator softmax: Shape inconsistent, Provided=(5,5418,1), inferred shape=(5,)")**

I also tried Reshape:
fc2  = mx.FullyConnected(act1, name=:fc2, num_hidden=100)
shape = mx.Reshape(fc2, shape=(batch_size, 5, 3))
softmax = mx.SoftmaxOutput(shape, label, name=:softmax)

And the output:
**/logging.h:300: [14:43:57] src/operator/tensor/./matrix_op-inl.h:137: Check failed: oshape.Size() == dshape.Size() (75012210 vs. 27090) Target shape size is different to source. Target: (2769,5418,5)
Source: (5,5418)**

(nvm the actual shape output)

**Comment by marugari** [comment]
I think
```
fc = mx.FullyConnected(num_hidden=5)
sm = mx.SoftmaxOutput(fc)
```
means multi-class (not multi-label) classification.

MXNet supports multi-output but it might be inefficient in multi-class classification.
https://github.com/dmlc/mxnet/issues/910

The following is instrumental.
https://github.com/dmlc/mxnet/issues/647
And my multi-input/multi-output example...
https://github.com/marugari/Notebooks/blob/master/mxnet_multi-input_multi-output.ipynb

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!

**Comment by ahkarami** [comment]
Hi, Is there any Multi-Label Classification code on MXNet?
